pandas/io/pytables.py

```diff
@@ -313,10 +313,14 @@ def read_hdf(path_or_buf, key=None, **kwargs):
 
         """
 
-    if kwargs.get('mode', 'a') not in ['r', 'r+', 'a']:
+    # Ignore any supplied mode, we only need to read data
```

Comment:
I just had another thought - even if we want to open in read-write mode because of the locking issues, the default should be 'r+' rather than 'a' so that trying to read a non-existent file gives a FileNotFoundError.